.github/workflows/scripts/kinds/verify-kinds.go

```diff
@@ -414,3 +311,113 @@ func contains(array []string, value string) bool {
 
 	return false
 }
+
+type kindRegistry struct {
+	zipDir string
+}
+
+func NewKindRegistry() (*kindRegistry, error) {
```

Comment:
What about splitting this implementation in two? One struct/function to download and unzip the registry archive from GitHub, and another one (the `KindRegistry` struct itself maybe?

Maybe overkill for such a small program, but it would have the nice side effect of not requiring us to open the archive whenever `findLatestDir()` or `getPublishedKind()` is called.

Reply:
Sorry I don't understand what would be the two parts of the implementation then, if the first one downloads and unzips the archive, what remains in the second part?